scripts/Engineering/texture/TextureUtils.py

```diff
@@ -0,0 +1,663 @@
+        additional_pars_to_fix = set(self.peak_params_to_fix) - set(pars_to_tie)
+        ties = []
+        for idom in range(1, function.nDomains()):
+            # tie global params to first
+            for par in pars_to_tie:
+                ties.append(f"f{idom}.f0.{par}=f0.f0.{par}")  # global peak pars
+            for ipar_bg in range(function[idom][1].nParams()):
+                par = function[idom][1].getParamName(ipar_bg)
+                ties.append(f"f{idom}.f1.{par}=f0.f1.{par}")
+            for par in additional_pars_to_fix:
+                # pars to be fixed but not global/already tied
+                function[idom][0].fixParameter(par)
+        # add ties as string (orders of magnitude quicker than self.function.tie)
+        return f"{str(function)};ties=({','.join(ties)})"
+
+
+def _get_run_and_prefix_from_ws_log(ws: Workspace2D, wsname: str) -> Tuple[str, str]:
+    try:
+        run = str(ws.getRun().getLogData("run_number").value)
+        prefix = wsname.split(run)[0]
+    except:
+        run = "unknown"
+        prefix = ""
+    return run, prefix
+
+
+def _get_grouping_from_ws_log(ws: Workspace2D) -> str:
+    try:
+        grouping = str(ws.getRun().getLogData("Grouping").value)
+    except RuntimeError:
+        grouping = "GROUP"
+    return grouping
+
+
+def fit_all_peaks(wss: Sequence[str], peaks: Sequence[float], peak_window: float, save_dir: str, override_dir: bool = False) -> None:
+    """
+    Fit all the peaks given in all the spectra of all the workspaces, for use in a texture analysis workflow
+
+    wss: Workspace names of all the workspaces to fit
+    peaks: Sequence of peak positions in d-spacing
+    peak_window: size of the window to create around the desired peak for purpose of fitting
+    save_dir: directory to save the results in
+    override_dir: flag which, if True, will save files directly into save_dir rather than creating a folder structure
+    """
+    for wsname in wss:
+        ws = ADS.retrieve(wsname)
+        run, prefix = _get_run_and_prefix_from_ws_log(ws, wsname)
+        grouping = _get_grouping_from_ws_log(ws)
+        for peak in peaks:
+            # change peak window to fraction
+            out_ws = f"{prefix}{run}_{peak}_{grouping}_Fit_Parameters"
+            out_file = out_ws + ".nxs"
+            out_path = path.join(save_dir, out_file) if override_dir else path.join(save_dir, grouping, str(peak), out_file)
+
+            xmin, xmax = peak - peak_window, peak + peak_window
+
+            out_tab = CreateEmptyTableWorkspace(OutputWorkspace=out_ws)
+
+            peak_func_name = "BackToBackExponential"
+            bg_func_name = "LinearBackground"
+
+            func_generator = TexturePeakFunctionGenerator([])  # don't fix any parameters
+            initial_function, md_fit_kwargs, intensity_estimates = func_generator.get_initial_fit_function_and_kwargs_from_specs(
+                ws, peak, (xmin, xmax), ("A", "B"), peak_func_name, bg_func_name
```

Comment:
I wonder whether it's worth exposing parameter to fix e.g., `("A","B")` to the user? From the fits it looks like you'd benefit from a second fit freeing these (or at least `B`) - but keeping them tied to the first peak.
<img width="400" height="300" alt="image" src="https://github.com/user-attachments/assets/edf0c147-0a14-4991-a567-f9e4e2dcaa0d" />
I'm still getting some workspaces with not a lot of data in them - maybe this is expected?
<img width="400" height="300" alt="image" src="https://github.com/user-attachments/assets/61dc6901-b6c6-4bb0-8855-96462bc0dec7" />
For these workspaces I would keep A,B fixed or tied - perhaps you can do an I/sigma check to see which spectra have a statistically significant peak?